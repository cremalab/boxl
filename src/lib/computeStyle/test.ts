import { computeStyle } from ".";

describe("computeStyle", () => {
  it("handles string", () => {
    const props = { style: "background: red;", theme: {} };
    const received = computeStyle(props);
    expect(received).toEqual(props.style);
  });

  it("handles tagged template", () => {
    const received = computeStyle({
      style: s => s`
        background: red;
        color: ${p => p.theme.color};
      `,
      theme: {
        color: "blue",
      },
    });
    expect(received).toMatchSnapshot();
  });
});
