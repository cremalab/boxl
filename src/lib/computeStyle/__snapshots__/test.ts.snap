// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`computeStyle handles tagged template 1`] = `
"background: red;
        color: blue;"
`;
